core/src/block_creation_loop.rs

```diff
@@ -317,10 +338,12 @@ fn record_and_complete_block(
         )?;
     }
 
-    // TODO: insert block footer
+    // Construct and send the block footer
+    let footer = produce_block_footer();
```

Comment:
awesome, this turned out very clean! FYI @akhi3030 you can plug in your rewards container here to populate the aggregates in the footer.

For the finalized certificate I imagine we can have something lightweight that can be updated on `VotorEvent::Finalized`. Can probably even live inside the rewards container and we can add it to the footer

Reply:
perfect, this looks nice and straightforward.